src/functor.jl

```diff
@@ -169,7 +169,54 @@ function _default_walk(f, x)
   func, re = functor(x)
   re(map(f, func))
 end
-_default_walk(f, ::Nothing, ::Nothing) = nothing
+_default_walk(_, ::Nothing, ::Nothing) = nothing
+
+# Side effects only, saves a restructure
+function _foreach_walk(f, x)
+  foreach(f, children(x))
+  return x
+end
```

Comment:
Can I suggest making:
```
function _structural_walk(f, x)
  func, re = functor(x)
  map(f, func)
end
```
This can be used by `fmapstructure` and `fcollect`, and an `fmapnothing` could just throw away the return.

Are there efficiency reasons to have a 3rd walk with explicit `foreach`? I guess if you have a non-leaf array, this will save allocating the result.

Reply:
Maybe I start to agree, and having just `walk = (f, x) -> map(f, children(x))` (as now) and `walk = (f, x) -> foreach(f, children(x))` around is better than giving these names. So maybe I retract the suggestion.